contracts/staking/contracts/src/libs/LibSafeDowncast.sol

```diff
@@ -57,4 +57,21 @@ library LibSafeDowncast {
         }
         return b;
     }
+
+    /// @dev Safely downcasts to a uint32
+    /// Note that this reverts if the input value is too large.
+    function downcastToUint32(uint256 a)
```

Comment:
Assuming [2143](https://github.com/0xProject/0x-monorepo/pull/2143) gets merged first pls add unit tests for this guy (should be ez copypasta)